Plugins/AWSLambdaPackager/Plugin.swift

```diff
@@ -0,0 +1,344 @@
+                .product(product.name),
+                parameters: parameters
+            )
+            guard result.builtArtifacts.count <= 1 else {
+                throw Errors.unknownExecutable("too many executable artifacts found for \(product.name)")
+            }
+            guard let artifact = result.builtArtifacts.first else {
+                throw Errors.unknownExecutable("no executable artifacts found for \(product.name)")
+            }
+            results[.init(product)] = artifact.path
+        }
+        return results
+    }
+
+    private func package(
+        products: [LambdaProduct: Path],
+        toolsProvider: (String) throws -> Path,
+        outputDirectory: Path,
+        verboseLogging: Bool
+    ) throws -> [LambdaProduct: Path] {
+        let zipToolPath = try toolsProvider("zip")
+
+        var archives = [LambdaProduct: Path]()
+        for (product, artifactPath) in products {
+            if verboseLogging {
+                print("-------------------------------------------------------------------------")
+                print("archiving \"\(product.name)\"")
+                print("-------------------------------------------------------------------------")
+            }
+
+            // prep zipfile location
+            let zipfilePath = outputDirectory.appending(product.name, "\(product.name).zip")
+            if FileManager.default.fileExists(atPath: zipfilePath.string) {
+                try FileManager.default.removeItem(atPath: zipfilePath.string)
+            }
+            try FileManager.default.createDirectory(atPath: zipfilePath.removingLastComponent().string, withIntermediateDirectories: true)
+
+            #if os(macOS) || os(Linux)
+            let arguments = ["--junk-paths", zipfilePath.string, artifactPath.string]
+            #else
+            throw Error.unsupportedPlatform("can't or don't know how to create a zipfile on this platform")
+            #endif
+
+            // run the zip tool
+            try self.execute(
+                executable: zipToolPath,
+                arguments: arguments,
+                verboseLogging: verboseLogging
+            )
+
+            archives[product] = zipfilePath
+
+            /*
+
+             target=".build/lambda/$executable"
+             rm -rf "$target"
+             mkdir -p "$target"
+             cp ".build/release/$executable" "$target/"
+             # add the target deps based on ldd
+             ldd ".build/release/$executable" | grep swift | awk '{print $3}' | xargs cp -Lv -t "$target"
+             cd "$target"
+             ln -s "$executable" "bootstrap"
+             zip --symlinks lambda.zip *
+
+             */
+            // docker run --rm -v "$workspace":/workspace -w /workspace/Examples/Deployment builder \
+            //               bash -cl "./scripts/package.sh $executable"
+        }
+        return archives
+    }
+
+    @discardableResult
+    private func execute(
```

Comment:
This kind of thing seems common enough that I wonder if PackagePlugins should provide a convenience cover.